src/main/java/ogc/rs/database/DatabaseServiceImpl.java

```diff
@@ -395,19 +396,15 @@ public Future<JsonObject> getFeature(String collectionId, Integer featureId, Map
                             // Feature exists, now apply filters
                             return applySpatialFiltersAndGetFeature(conn, collectionId, featureId, limits, geoColumn, collector);
                         })
-        ).onComplete(ar -> {
-            if (ar.succeeded()) {
-                result.complete(ar.result());
-            } else {
-                Throwable cause = ar.cause();
-                if (cause instanceof OgcException) {
-                    result.fail(cause);
+        ).onSuccess(success -> result.complete(success))
+        .onFailure(fail -> {
+                if (fail instanceof OgcException) {
```

Comment:
Will this ever be an instance of OgcException? I think it'll always be a PgException.